### PR TITLE
Support signing sigv4 requests

### PR DESCRIPTION
**Pull request by alichay**
Allows signing Sigv4 requests (such as S3 requests).
Expects a "valid" sigv4 request in, but does not validate the access or secret keys (the only validation done is via tokenizer), it just extracts region, service, and the request timestamp from the "Authorization" header for its own signing.

### CODE REVIEW

File: processor.go [hunk 1]
```diff
@@ -145,6 +148,77 @@ func (c *OAuthProcessorConfig) Processor(params map[string]string) (RequestProce
...
+				if err != nil {
+					return err
+				}
+
+				service = credParts[2]
+				region = credParts[3]
+				break
+			}
+		}
+
```
Comment (btoews) [inline processor.go:208]: Should we check here that we found a good credential in the header or is it okay for `service`, `region`, and `date` to be empty?

**Comment by btoews** [comment]
I deployed this branch to tokenizer.fly.io if that's helpful for testing your other changes.

File: processor.go (hunk 1 above)
Comment (alichay) [reply processor.go:214]: great catch, I missed the error handling in the refactor 😅

**Review by btoews — APPROVED** [review]
LGTM

**Pull request merged**